a11y and usability fixes

akash07k opened the pull request:
Lots of accessibility and usability fixes. #1133

akash07k commented:
I don't know why, but when I'm expanding the sources, the sources are not displaying.

akash07k commented on templates/home.phtml:
```diff
@@ -111,11 +111,11 @@
 
     <div id="loginform">
-        <form action="<?= $this->base; ?>" method="post">
+        <form action="<?= $this->base; ?>" method="post" autocomplete="on">
```
> Hmm, should not this be on by default. Quick read of https://developer.mozilla.org/en-US/docs/Web/Security/Securing_your_site /Turning_off_form_autocompletion seems to suggest that it is the case.
> 
> Hmm, I did not know about [`autocomplete="new-password"`](https://developer.mozilla.org/en-US/docs/Web/Security/Securing_your_site/Turning_off_form_autocompletion#Preventing_autofilling_with_autocompletenew-password), that will be useful for the password generator page.
Quoting from the same URL:
It is important to know that if you turn off autocomplete, you are breaking the rule 1.3.5: Identify Input Purpose in WCAG 2.1. So if you are making a website that should follow WCAG you should use autocomplete with autofill instead for to the users advantage. 
Please refer:
https://www.w3.org/WAI/WCAG21/Understanding/identify-input-purpose.html

akash07k replied:
> > Hmm, should not this be on by default. Quick read of https://developer.mozilla.org/en-US/docs/Web/Security/Securing_your_site /Turning_off_form_autocompletion seems to suggest that it is the case.
> > Hmm, I did not know about [`autocomplete="new-password"`](https://developer.mozilla.org/en-US/docs/Web/Security/Securing_your_site/Turning_off_form_autocompletion#Preventing_autofilling_with_autocompletenew-password), that will be useful for the password generator page.
> > Quoting from the same URL:
> > It is important to know that if you turn off autocomplete, you are breaking the rule 1.3.5: Identify Input Purpose in WCAG 2.1. So if you are making a website that should follow WCAG you should use autocomplete with autofill instead for to the users advantage.
> > Please refer:
> > https://www.w3.org/WAI/WCAG21/Understanding/identify-input-purpose.html

Though, I'll add autocomplete="new-password" to the hashpassword templet

jtojnar replied:
Yeah, I know about

>  It is important to know that if you turn off autocomplete, you are breaking the rule 1.3.5: Identify Input Purpose in WCAG 2.1. So if you are making a website that should follow WCAG you should use autocomplete with autofill instead for to the users advantage.

But we are not turning it off as far as I can tell.

akash07k replied:
> Yeah, I know about
> 
> > It is important to know that if you turn off autocomplete, you are breaking the rule 1.3.5: Identify Input Purpose in WCAG 2.1. So if you are making a website that should follow WCAG you should use autocomplete with autofill instead for to the users advantage.
> 
> But we are not turning it off as far as I can tell.

Yes, but we should explicitly turn it on. rest, I'll do whatever you say.
If you want, then I'll remove it.

jtojnar replied:
> Yes, but we should explicitly turn it on.

Why? Does some screen reader not support the default value?

The [spec](https://www.w3.org/TR/html52/sec-forms.html#ref-for-form-owner%E2%91%A3%E2%91%A5) says:

> If the autocomplete attribute is omitted, the default value corresponding to the state of the element’s form owner’s autocomplete attribute is used instead (either "on" or "off"). If there is no form owner, then the value "on" is used.

akash07k replied:
> > Yes, but we should explicitly turn it on.
> 
> Why? Does some screen reader not support the default value?
> 
> The [spec](https://www.w3.org/TR/html52/sec-forms.html#ref-for-form-owner%E2%91%A3%E2%91%A5) says:
> 
> > If the autocomplete attribute is omitted, the default value corresponding to the state of the element’s form owner’s autocomplete attribute is used instead (either "on" or "off"). If there is no form owner, then the value "on" is used.

Almost all the screen readers do support the default value, but there are some issues with the recent builds of Narrator which ships by default with Windows and I've added this explicitly to work around that. But the market share for Narrator is quite low, so I can remove it if you want.

akash07k commented on templates/home.phtml:
```diff
@@ -174,22 +174,22 @@
             <?php endif; ?>
 
-            <a id="nav-copyright" href="https://selfoss.aditu.de" target="_blank" rel="noopener noreferrer">selfoss <?= \F3::get('version'); ?></a>
+            <a id="nav-copyright" href="https://selfoss.aditu.de" target="_blank" rel="noopener noreferrer"><?= trim(\F3::get('html_title')); ?> <?= \F3::get('version'); ?></a>
```
I agree with you, but since URL leads the user to the original selfoss website, the name customisation should be fine.
Rest, it is your choice, but I'll request you to include this customisation for the better user experience.

akash07k replied:
> I think this is an information about the version of the software so it should probably always be selfoss, rather than the name administrator chooses for a given installation. Though I would argue that user should not need to care about what program this really is so I would move it to the settings.

Actually, I think for the better user experience, the name should be retrieved from the settings itself.
Also, since URL is unchanged and leads the user to the selfoss homepage, it should be fine. rest, it is your choice.

jtojnar replied:
> Also, since URL is unchanged and leads the user to the selfoss homepage, it should be fine.

Well this is precisely my point. I would expect the label to match the target URL so that user knows where will it lead before visiting it.

Also it is followed by a selfoss version number, which, I think, does not make sense when combined with a custom title.

I would just remove this link altogether, as it does not add much value while cluttering the sidebar.

akash07k replied:
> > Also, since URL is unchanged and leads the user to the selfoss homepage, it should be fine.
> 
> Well this is precisely my point. I would expect the label to match the target URL so that user knows where will it lead before visiting it.
> 
> Also it is followed by a selfoss version number, which, I think, does not make sense when combined with a custom title.
> 
> I would just remove this link altogether, as it does not add much value while cluttering the sidebar.

Yeah, removing is also a better option.
It will be great.
Also, when will you commit your changes so that I can pull them and work further?
I also want to introduce a heading level1 on th home page because user always looks for the main heading I.E: h1 for reading about the website/application and for the main content. Since main content is already present, where and with which content we should add h1?

akash07k commented on .gitignore:
```diff
@@ -7,6 +7,7 @@
 /data/sqlite/*.db
 /public/all.*
 /public/selfoss-sw-offline.js
+/public/package-lock.json
```
Are you refering to .gitignore?

akash07k replied:
> We should start committing this file.

Are you refering to .gitignore?

jtojnar replied:
> Are you refering to .gitignore?

Yes. Since selfoss is an application, we should pin the dependencies uing the lock file. I will commit

jtojnar replied:
> Are you refering to .gitignore?

Yes. Since selfoss is an application, we should pin the dependencies uing the lock file. I will commit it in https://github.com/SSilence/selfoss/pull/1096, which deals with changing the dependencies.

akash07k replied:
> > Are you refering to .gitignore?
> 
> Yes. Since selfoss is an application, we should pin the dependencies uing the lock file. I will commit it in #1096, which deals with changing the dependencies.

So, what should I do as of now?

jtojnar replied:
I would just revert this commit for now.

akash07k replied:
> I would just revert this commit for now.

Ok, no problem.

akash07k commented on templates/home.phtml:
```diff
@@ -141,27 +141,27 @@
             <button id="nav-mark"><?= \F3::get('lang_markread')?></button>
 
             <div id="nav-filter-wrapper">
-            <h2 id="nav-filter-title" class="nav-filter-expanded" tabindex="0" aria-haspopup="true" aria-expanded="true"><?= \F3::get('lang_filter') ?></h2>
```
Hmm, so what I'm supposed to do here? aria-haspopup is not required in this case.

akash07k replied:
> Hmm, I wonder why I added haspopup in [dd728f5](https://github.com/SSilence/selfoss/commit/dd728f5e1ba32eab3f28fbbb4abdda7f3822065e)

Hmm, so what I'm supposed to do here? aria-haspopup is not required in this case.

jtojnar replied:
Yeah, removing it was a good catch.

akash07k commented on templates/home.phtml:
```diff
@@ -141,27 +141,27 @@
-                <li id="nav-filter-unread" class="nav-filter-unread" role="link" tabindex="0"><?= \F3::get('lang_unread')?>
+            <h2 id="nav-filter-title" class="nav-filter-expanded" tabindex="0" aria-expanded="true"><span role="link"><?= \F3::get('lang_filter') ?></span></h2>
+            <ul id="nav-filter">
+                <li id="nav-filter-newest" class="nav-filter-newest" role="link" tabindex="0"><span role="listitem"><?= \F3::get('lang_newest')?> <span class="offline-count offlineable" title="<?= \F3::get('lang_offline_count')?>"></span><span class="count" title="<?= \F3::get('lang_online_count')?>"></span></span></li>
```
I was also thinking like that and was thinking to use <a> instead of it. but I thaught that may be you'll disagree with due to design constraints.

akash07k replied:
<a> ancor tag

akash07k replied:
> I wonder if it is alright to have listitems inside links. We should probably use `<a>` inside the `<li>`.

I was also thinking like that and was thinking to use ancor tag instead of it. but I thaught that may be you'll disagree with due to design constraints.

jtojnar replied:
Design can be updated. Accessibility and clean code should come first.

akash07k replied:
> Design can be updated. Accessibility and clean code should come first.

Thanks, you are great. Noone considers accessibility seriously and everybody looks at design. You are really caring about accessibility and this means a lot to me. Thanks a tons my dear bro.

akash07k replied:
> I wonder if it is alright to have listitems inside links. We should probably use `<a>` inside the `<li>`.

Yes, we should have `<a>` inside of `<li>` but when I'm trying it, it is breaking badly. don't know what exactly is the issue. Tried several times but no luck. Can you please try it at your end? by doing so, there won't be any need of explicit `role="listitem"` because nesting `<a>` inside of the `<li>` will automaticly fix everything regarding the readability of the list items.

akash07k commented on templates/home.phtml:
```diff
@@ -157,11 +157,11 @@
                 <li class="active nav-tags-all" role="link" tabindex="0"><span role="listitem"><?= \F3::get('lang_alltags')?></span></li>
             </ul>
-            <h2 id="nav-sources-title" class="nav-sources-collapsed" tabindex="0" aria-expanded="false"><span role="link"><?= \F3::get('lang_sources') ?></span></h2>
+            <h2 id="nav-sources-title" class="nav-sources-collapsed" tabindex="0"><span role="link"  aria-expanded="false"><?= \F3::get('lang_sources') ?></span></h2>
```
> This might require JavaScript change.

Hmm, so, can you please do the required changes?

jtojnar replied:
Sure. Will update the scripts and the styles later today.

akash07k commented on public/js/selfoss-events-entriestoolbar.js:
```diff
@@ -49,10 +49,10 @@ selfoss.events.entriesToolbar = function(parent) {
         if (parent.find('.entry-toolbar').has('button.entry-share' + shares[0]).length == 0) {
             // add the share toolbar entries
             parent.find('.entry-smartphone-share button.entry-newwindow').after(selfoss.shares.buildLinks(shares, function(name) {
-                return '<button class="entry-share entry-share' + name + '" title="' + name + '"><img class="entry-share" title="' + name + '" src="images/' + name + '.png" height="16" width="16">' + name + '</button>';
```
> Why use `aria-haspopup`? It opens a new tab/window, not a popup.

You are right, but, since it behaves like link, either it should be changed to link, or if it is required to keep it button, then aria-haspopup is best bet for this. Since, normal buttons are not supposed to open any new tab/page

jtojnar replied:
I will defer to you on this. Unfortunately, we cannot use link, as some sharers work in the background (e.g. copy to clipboard, share with operating system menu).

Maybe we could improve the button labels.

akash07k replied:
> I will defer to you on this. Unfortunately, we cannot use link, as some sharers work in the background (e.g. copy to clipboard, share with operating system menu).
> 
> Maybe we could improve the button labels.

No problem, we can improve the labels.
Should I do something or you're already doing?

jtojnar replied:
I will push the changes shortly.

akash07k commented on templates/home.phtml:
```diff
@@ -141,8 +141,8 @@
             <div id="nav-filter-wrapper">
-            <h2 id="nav-filter-title" class="nav-filter-expanded" tabindex="0"><span role="link"  aria-expanded="true"><?= \F3::get('lang_filter') ?></span></h2>
-            <ul id="nav-filter">
+            <span role="heading" aria-level="2"><h2 id="nav-filter-title" class="nav-filter-expanded" tabindex="0" role="link" aria-expanded="true"><?= \F3::get('lang_filter') ?></h2></span>
```
> This is not valid HTML. We should make it a `<button>` inside `<h2>`.

I'll change appropriately.

akash07k replied:
> This is not valid HTML. We should make it a `<button>` inside `<h2>`.

I've just tried that but it doesn't work.
Because the class name is coded in the js file and thus we have to use aria-expanded="true" on the heading only.
And if we do not use role="link" then aria-expanded doesn't work for headings.
Thats why I used this hack.
If you can change somethings in order to fix it, then it will be very helpful.

akash07k commented on templates/home.phtml:
```diff
@@ -141,8 +141,8 @@
-            <h2 id="nav-filter-title" class="nav-filter-expanded" tabindex="0"><span role="link"  aria-expanded="true"><?= \F3::get('lang_filter') ?></span></h2>
-            <ul id="nav-filter">
+            <span role="heading" aria-level="2"><h2 id="nav-filter-title" class="nav-filter-expanded" tabindex="0" role="link" aria-expanded="true"><?= \F3::get('lang_filter') ?></h2></span>
+            <ul id="nav-filter" aria-label="<?= \F3::get('lang_filters_list_label') ?>">
```
> Would it make sense to use `aria-labelledby`?

aria-label is fine here since for aria-labeledby we are required to show/define the corresponding string somewhere on the page and then we will have to retrieve the ID of it.

jtojnar replied:
Also why did you create new strings? Or would it make sense to make them more verbose for the screen reader? Like “Filter by source”

akash07k replied:
> Also why did you create new strings? Or would it make sense to make them more verbose for the screen reader? Like “Filter by source”

Actually I had to create the new strings in order to asign the labels for the lists.
And this makes retrieving the lists very pleasant by the screen reader users because just by the label only they can recognise that what the list is for.
Implementing the aria-label for the lists is very good and leaves a good impression on the user about their navigation experience.

akash07k commented on templates/item.phtml:
```diff
@@ -27,12 +27,12 @@
     <a href="<?= \helpers\Anonymizer::anonymize($this->item['link']); ?>" class="entry-icon" tabindex="-1" rel="noopener noreferrer" aria-hidden="true">
         <?php if (strlen(trim($this->item['icon'])) > 0 && $this->item['icon'] != '0') : ?>
-        <img src="<?= 'favicons/' . $this->item['icon']; ?>" alt="" >
+        <img src="<?= 'favicons/' . $this->item['icon']; ?>" aria-hidden="true" alt="" >
```
> Weird, does not the `aria-hidden` get inherited from the link?

It gets inherited, but some screen readers still do not respond to it properly.

akash07k commented on templates/home.phtml:
```diff
@@ -170,26 +170,26 @@
-            <input aria-label="<?= \F3::get('lang_search_label') ?>" type="search" id="search-term">
-            <button id="search-remove" title="<?= \F3::get('lang_searchremove')?>" aria-label="<?= \F3::get('lang_searchremove')?>"><img src="images/remove.png" alt=""></button>
-            <button id="search-button" title="<?= \F3::get('lang_searchbutton')?>" aria-label="<?= \F3::get('lang_searchbutton')?>"><img src="images/search.png" alt=""></button>
+            <input aria-label="<?= \F3::get('lang_search_label') ?>" type="text" id="search-term" accesskey="s">
```
> Why not `type="search"`.

Yes, "search" is fine, I'll revert it.

akash07k commented:
So, should I wait for your commits in order to fix things further?

On 9/23/19, Jan Tojnar <notifications@github.com> wrote:
> jtojnar commented on this pull request.
>
>
>
>>              <ul id="nav-tags">
>                  <li class="active nav-tags-all" role="link"
> tabindex="0"><span role="listitem"><?=
> \F3::get('lang_alltags')?></span></li>
>              </ul>
> -            <h2 id="nav-sources-title" class="nav-sources-collapsed"
> tabindex="0" aria-expanded="false"><span role="link"><?=
> \F3::get('lang_sources') ?></span></h2>
> +            <h2 id="nav-sources-title" class="nav-sources-collapsed"
> tabindex="0"><span role="link"  aria-expanded="false"><?=
> \F3::get('lang_sources') ?></span></h2>
>
> Sure. Will update the scripts and the styles later today.
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/SSilence/selfoss/pull/1134#discussion_r327069113

akash07k commented:
> I don't know why, but when I'm expanding the sources, the sources are not displaying.

They are displaying now, but screen reader is not treating them as list still after changing the code.

jtojnar commented:
Feel free to address what you like, I will be back from school in two or three hours and then will take a jab at it.

akash07k commented:
> Feel free to address what you like, I will be back from school in two or three hours and then will take a jab at it.

Sure, also, if you find anything wrong in the code, try to remove my changes.

akash07k commented:
Ok, thanks

On 9/23/19, Jan Tojnar <notifications@github.com> wrote:
> jtojnar commented on this pull request.
>
>
>
>> @@ -49,10 +49,10 @@ selfoss.events.entriesToolbar = function(parent) {
>          if (parent.find('.entry-toolbar').has('button.entry-share' +
> shares[0]).length == 0) {
>              // add the share toolbar entries
>              parent.find('.entry-smartphone-share
> button.entry-newwindow').after(selfoss.shares.buildLinks(shares,
> function(name) {
> -                return '<button class="entry-share entry-share' + name + '"
> title="' + name + '"><img class="entry-share" title="' + name + '"
> src="images/' + name + '.png" height="16" width="16">' + name +
> '</button>';
>
> I will push the changes shortly.
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/SSilence/selfoss/pull/1134#discussion_r327163030

akash07k commented:
I've just noticed that already there are strings for those texts.

jtojnar commented:
Yup, they are in the headings above, which is why I suggested to use `aria-labelledby`. But if it is better for you to have the label be `Filter by status/tag/source` (a context that seeing users will get from the filters being together in the sidebar), it might be better to use a separate string.

Though, I would also be worried the screen reader will now read the label twice:

```
[Heading level 3] Tags
[List] Tags
…
```

akash07k commented:
> Yup, they are in the headings above, which is why I suggested to use `aria-labelledby`. But if it is better for you to have the label be `Filter by status/tag/source` (a context that seeing users will get from the filters being together in the sidebar), it might be better to use a separate string.
> 
> Though, I would also be worried the screen reader will now read the label twice:
> 
> ```
> [Heading level 3] Tags
> [List] Tags
> …
> ```

Pushed the changes. removed the duplicated strings and used aria-labeledby
No. there's no issue, for example, JAWS reads like this:
Heading level 2 tags.
Tags list of 19 items.
Which is fine.

jtojnar commented:
* Dropped the db patches since that is separate issue (#944)
* Converted the `<span>`s with roles to proper HTML elements, fixed the JavaScript and styles.
* Specialized the login form `autocomplete` attributes.
* Removed the branding link in the sidebar.

I left out the `aria-expanded` on `.entry-title`, as that one is non-trivial to fix. We will need the closing/opening methods as @niol suggested in https://github.com/SSilence/selfoss/issues/1127#issuecomment-524739770.

jtojnar commented:
I am looking at adding the buttons to the entry titles some more and there is an issue: micro-articles such as tweets can contain links inside the titles. For that reason we cannot really make the whole title clickable. I am not sure how to represent that properly.

jtojnar commented:
For now, I have made the `.entry-title` a `role=link` and moved `aria-expanded` there. But I am not sure how will it behave with tweets containing links in title. Here is an example of such tweet body, could you check how it behaves with screen readers https://jsfiddle.net/2hxLrgjt/1/show.

I have also rebased this onto master and it is ready to be merged, pending a review from @niol regarding the latest commit.

akash07k commented:
> For now, I have made the `.entry-title` a `role=link` and moved `aria-expanded` there. But I am not sure how will it behave with tweets containing links in title. Here is an example of such tweet body, could you check how it behaves with screen readers https://jsfiddle.net/2hxLrgjt/1/show.
> 
> I have also rebased this onto master and it is ready to be merged, pending a review from @niol regarding the latest commit.

I've just checked your example, no problem, screen readers are recognising every link as individual link and there's no problem with the readability also.
I feel that since screen readers are not having any problem with the links contained in the title, can't we just make the entry title button as before? I don't think that it will cause problems.

akash07k commented on templates/tag.phtml:
```diff
@@ -1,7 +1,7 @@
 <li role="link" tabindex="0">
-    <span class="tag"><?= $this->tag; ?></span>
+    <span role="listitem"><span class="tag"><?= $this->tag; ?></span>
```
> `span` opened twice?

I can't see the duplicated span opening or closing in recent repo, I think it was previously and resolved by @jtojnar

jtojnar replied:
Yeah, I indented it to make the nesting cleaner.

akash07k commented:
> I am looking at adding the buttons to the entry titles some more and there is an issue: micro-articles such as tweets can contain links inside the titles. For that reason we cannot really make the whole title clickable. I am not sure how to represent that properly.

If we'll make the whole title clickable, won't the user be able to click the individual links contained in the title?

jtojnar commented:
> I've just checked your example, no problem, screen readers are recognising every link as individual link and there's no problem with the readability also.

That’s great.

> I feel that since screen readers are not having any problem with the links contained in the title, can't we just make the entry title button as before? I don't think that it will cause problems.

Unfortunately, HTML does not allow interactive elements inside [links](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-a-element:concept-element-content-model) or [buttons](https://html.spec.whatwg.org/#the-button-element:concept-element-content-model), so we cannot use the native elements. I would say the `link` role fits better than the `button`, since expanding the item also changes the URL hash to point to that page.

> If we'll make the whole title clickable, won't the user be able to click the individual links contained in the title?

Not if we use native HTML elements.

akash07k commented:
oh, ok. fine. let's stick with role="link" as of now. if there will be
any issue, then we'll look forward for role="button"

On 9/24/19, Jan Tojnar <notifications@github.com> wrote:
>> I've just checked your example, no problem, screen readers are recognising
>> every link as individual link and there's no problem with the readability
>> also.

>

> That’s great.

>

>> I feel that since screen readers are not having any problem with the links
>> contained in the title, can't we just make the entry title button as
>> before? I don't think that it will cause problems.

>

> Unfortunately, HTML does not allow interactive elements inside
> [links](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-a-element:concept-element-content-model)
> or
> [buttons](https://html.spec.whatwg.org/#the-button-element:concept-element-content-model),
> so we cannot use the native elements. I would say the `link` role fits
> better than the `button`, since expanding the item also changes the URL hash
> to point to that page.

>

>> If we'll make the whole title clickable, won't the user be able to click
>> the individual links contained in the title?

>

> Not if we use native HTML elements.

>

> --

> You are receiving this because you authored the thread.

> Reply to this email directly or view it on GitHub:

> https://github.com/SSilence/selfoss/pull/1134#issuecomment-534641301

akash07k commented:
Is there any way to identify that which filter, source and tag is
currently active and put aria-current on it?

On 9/24/19, Akash Kakkar <akash.galaxy07@gmail.com> wrote:
> oh, ok. fine. let's stick with role="link" as of now. if there will be
> any issue, then we'll look forward for role="button"
>
> On 9/24/19, Jan Tojnar <notifications@github.com> wrote:
>>> I've just checked your example, no problem, screen readers are
>>> recognising
>>> every link as individual link and there's no problem with the
>>> readability
>>> also.
>
>>
>
>> That’s great.
>
>>
>
>>> I feel that since screen readers are not having any problem with the
>>> links
>>> contained in the title, can't we just make the entry title button as
>>> before? I don't think that it will cause problems.
>
>>
>
>> Unfortunately, HTML does not allow interactive elements inside
>> [links](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-a-element:concept-element-content-model)
>> or
>> [buttons](https://html.spec.whatwg.org/#the-button-element:concept-element-content-model),
>> so we cannot use the native elements. I would say the `link` role fits
>> better than the `button`, since expanding the item also changes the URL
>> hash
>> to point to that page.
>
>>
>
>>> If we'll make the whole title clickable, won't the user be able to click
>>> the individual links contained in the title?
>
>>
>
>> Not if we use native HTML elements.
>
>>
>
>> --
>
>> You are receiving this because you authored the thread.
>
>> Reply to this email directly or view it on GitHub:
>
>> https://github.com/SSilence/selfoss/pull/1134#issuecomment-534641301
>

jtojnar commented:
Good idea. Will do that.

akash07k commented:
Thanks bro.

On 9/24/19, Jan Tojnar <notifications@github.com> wrote:
> Good idea. Will do that.
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/SSilence/selfoss/pull/1134#issuecomment-534651935

akash07k commented:
I'm waiting for the pagination implementation and will try to fix some
things if required in that as well

On 9/24/19, Akash Kakkar <akash.galaxy07@gmail.com> wrote:
> Thanks bro.
>
> On 9/24/19, Jan Tojnar <notifications@github.com> wrote:
>> Good idea. Will do that.
>>
>> --
>> You are receiving this because you authored the thread.
>> Reply to this email directly or view it on GitHub:
>> https://github.com/SSilence/selfoss/pull/1134#issuecomment-534651935
>

jtojnar commented:
@niol I refactored entry selection as well. Could you please take a look at it as well?

jtojnar commented:
I think the `aria-current` for navigation will be better handled in https://github.com/SSilence/selfoss/pull/1064

akash07k commented:
Thanks a lot for this, but now aria-expanded is not working for the
entry titles.
When we are clicking on an entry title, expanded="true" is not
happening. can you please look at this?

On 9/25/19, Jan Tojnar <notifications@github.com> wrote:
> I think the `aria-current` for navigation will be better handled in
> https://github.com/SSilence/selfoss/pull/1064
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/SSilence/selfoss/pull/1134#issuecomment-534770920

akash07k commented:
Thanks for your commits bro, it's working properly.
No issue so far.

On 9/25/19, Akash Kakkar <akash.galaxy07@gmail.com> wrote:
> Thanks a lot for this, but now aria-expanded is not working for the
> entry titles.
> When we are clicking on an entry title, expanded="true" is not
> happening. can you please look at this?
>
> On 9/25/19, Jan Tojnar <notifications@github.com> wrote:
>> I think the `aria-current` for navigation will be better handled in
>> https://github.com/SSilence/selfoss/pull/1064
>>
>> --
>> You are receiving this because you authored the thread.
>> Reply to this email directly or view it on GitHub:
>> https://github.com/SSilence/selfoss/pull/1134#issuecomment-534770920
>

jtojnar commented:
I have pushed the composer updates to master as there do not seem to be any breaking changes.

As for the jQuery update, the security vulnerabilities [do not affect us](https://github.com/SSilence/selfoss/issues/1029#issuecomment-376120050) and there are many breaking changes regarding to [`Deferred`](https://jquery.com/upgrade-guide/3.0/#deferred), which will need to be manually checked. And we should probably switch to `A+`-style promises throughout the codebase.

The remaining deprecations and breaking changes should be addressed in 8e991c4c9b8c2bd52b7c227295494a4b4c6a50a6, except for [`bind()` deprecation](https://jquery.com/upgrade-guide/3.0/#deprecated-bind-and-delegate). That is better left for a separate PR.

jtojnar commented:
Thanks for all the help.

akash07k commented:
> Thanks for all the help.

Thanks a lot to you too and @niol As well. I'm really liking contributing with you people and I'm very happy with all your cooperation.
You guys are really great.
Also, if I have some other fix for a11y, should I open a new pull requestor reopen it with my changes? What do you recommend?

jtojnar commented:
Please open a new pull request. GitHub does not allow re-opening PRs that were merged anyway.

Also please try to split commits that do several (even small) different things on different components into multiple commits. And use `git commit --fixup` when you need to fix something in a previous commit – that will allow us to easily squash typo commits before the pull request is merged.

akash07k commented:
> Please open a new pull request. GitHub does not allow re-opening PRs that were merged anyway.
> 
> Also please try to split commits that do several (even small) different things on different components into multiple commits. And use `git commit --fixup` when you need to fix something in a previous commit – that will allow us to easily squash typo commits before the pull request is merged.

Sure!

jtojnar commented:
Hi @akash07k. Previously, it [was not clear](https://github.com/fossar/selfoss/issues/1218) if selfoss was licensed under GPL 3 only, or also any later version. Could you clarify whether you are fine with licensing your contribution under `GPL-3-or-later`?

Thanks again and sorry for the confusion.

akash07k commented:
@jtojnar 
Yes, absolutely buddy.
No problem at all

> 
> 
> Hi @akash07k. Previously, it [was not clear](https://github.com/fossar/selfoss/issues/1218) if selfoss was licensed under GPL 3 only, or also any later version. Could you clarify whether you are fine with licensing your contribution under `GPL-3-or-later`?
> 
> Thanks again and sorry for the confusion.
